### ISSUE TITLE
Blog "series"

### ISSUE THREAD

**Issue by david-driscoll** [statiqdev/Statiq#512]
It would be awesome of the Blog Recipe had the ability to add a table of contents like piece for all posts in a given series.

I'll probably work this into my config / pipeline set and once I'm happy with it make a PR, but for now this is just to see if it's a cool idea or not. 👍

**Comment by daveaglick** [comment]
Love this idea - got a couple series myself that would benefit.

**Comment by adamclifford** [comment]
How would this be available for a theme that doesn't have a table of contents layout included?

**Comment by daveaglick** [comment]
It probably wouldn't. Maybe on the post itself "Other items in this series".

In general we're kind of settling on an approach for recipes where the recipe includes many settings for common scenarios, but themes can choose to ignore or not support all of them. I've been struggling with the balance here. On the one hand, we can take the Jekyll approach of not really having the recipe do much beyond the basics and leaving that up to each theme to add extras. The downside is you get much less theme portability - and while it makes sense for Jekyll since that's all it does, I feel like Wyam recipes should at least be a little more specialized.

On the other hand we could go too far and add features that are too niche to the recipe and burden theme authors. I think we've got a pretty good balance right now and I don't want to add too much more directly to the recipe at this point. I'm fine with themes not supporting everything the recipe has to offer and I've been trying to design new recipe features with this in mind (I.e., build will still work fine if the theme doesn't know or care about the feature).

**Comment by TechWatching** [comment]
Is there something planned for blog "series" with Statiq ?

**Comment by daveaglick** [comment]
Yeah, I'd love to get some sort of series feature added. As mentioned above (and now baked into the Statiq way of doing things), the theme is what would be responsible for implementing this feature. Since there's only one blogging theme right now with [CleanBlog](https://github.com/statiqdev/CleanBlog), adding it there would make it easier to copy to future themes.

All that said, it's not at the top of my current priority list given the need to get Statiq Docs out the door and push all the projects over to 1.0. I suspect adding series support to CleanBlog (and adding additional themes as well) will likely come after that's complete, so I'm not sure how long it will be before I get to it.
